File: Serene3/Serene3.Web/Modules/ServerTypes/com/countriesRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface countriesRow {
    Id?: number;
    Name?: string;
}

export abstract class countriesRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'com.countries';
    static readonly deletePermission = 'countries';
    static readonly insertPermission = 'countries';
    static readonly readPermission = 'countries';
    static readonly updatePermission = 'countries';

    static readonly Fields = fieldsProxy<countriesRow>();
}

File: Serene3/Serene3.Web/Modules/com/countries/countriesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Serene3.com;

[ConnectionKey("cetgeroycontrol"), Module("com"), TableName("Countries")]
[DisplayName("Countries"), InstanceName("Countries")]
[ReadPermission("countries")]
[ModifyPermission("countries")]
[ServiceLookupPermission("countries")]
public sealed class countriesRow : Row<countriesRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }

    [DisplayName("Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;

    }
}